src/main/scala/pricemigrationengine/services/ZuoraLive.scala

```diff
@@ -0,0 +1,85 @@
+package pricemigrationengine.services
+
+import java.time.LocalDate
+
+import pricemigrationengine.model._
+import scalaj.http.{Http, HttpOptions, HttpRequest}
+import upickle.default._
+import zio.{ZIO, ZLayer}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+
+object ZuoraLive {
+  val impl: ZLayer[Configuration, ConfigurationFailure, Zuora] = ZLayer.fromServiceM { configuration =>
+    configuration.config map { config =>
+      new Zuora.Service {
+
+        private val readTimeout = HttpOptions.readTimeout(Duration(30, SECONDS).toMillis.toInt)
+
+        private case class AccessToken(access_token: String)
+        private implicit val rwAccessToken: ReadWriter[AccessToken] = macroRW
+
+        private case class InvoicePreviewRequest(accountId: String, targetDate: LocalDate)
+        private implicit val rwInvoicePreviewRequest: ReadWriter[InvoicePreviewRequest] = macroRW
+
+        private lazy val fetchedAccessToken: ZIO[Any, ZuoraFetchFailure, String] =
```

Comment:
I dont think this lazy val will have the expected effect ie the fetchAccessToken effect will get executed every time its referenced.

You could fetch the access token in the constructor.

There is an IO.memoize function, Im not 100% sure how to uses it, ie it gives you an nested IO which is a bit weird.

Reply:
That's a good point.  I can't make head or tail of how `memoize` is supposed to work; there's also a `once` method which is similarly opaque.  I think I'll just create it in the conventional way, as you suggest.